clippy_lints/src/redundant_async_block.rs

```diff
@@ -0,0 +1,51 @@
+use clippy_utils::{diagnostics::span_lint_and_sugg, source::snippet};
+use rustc_ast::ast::*;
+use rustc_errors::Applicability;
+use rustc_lint::{EarlyContext, EarlyLintPass};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for `async` block that only returns `await` on a future.
+    ///
+    /// ### Why is this bad?
+    /// It is shorter to directly use the future.
```

Comment:
I'm curious: Does the `async await` really get reduced to a single call, or even inlined in release mode? If not, there would even be a perf angle.

Reply:
I call still see the await process in the MIR expansion in release mode, so I don't think this is optimized away.